api-test/internal/e2e/crypto.go

```
@@ -0,0 +1,229 @@
+	"github.com/mosip/esignet/api-test/internal/httpx"
+)
+
+// b64 is base64url without padding, the JOSE/JWT encoding.
+func b64(b []byte) string { return base64.RawURLEncoding.EncodeToString(b) }
+
+// generateRSA makes a fresh 2048-bit RSA key for the throwaway test client.
+func generateRSA() (*rsa.PrivateKey, error) { return rsa.GenerateKey(rand.Reader, 2048) }
+
+// publicJWK renders the public half as a JWK object (for client registration).
+func publicJWK(priv *rsa.PrivateKey, kid string) map[string]any {
+	pub := priv.Public().(*rsa.PublicKey)
```

golangci-lint (2.12.2)

[error] type assertion must be checked

(forcetypeassert)